tests/integration/suite/test_clustertemplate.py

```diff
@@ -479,6 +479,77 @@ def test_required_template_question(admin_mc, remove_resource):
         assert e.value.error.status == 422
 
 
+def test_secret_template_answers(admin_mc, remove_resource):
+    cluster_template = create_cluster_template(admin_mc,
+                                               remove_resource, [], admin_mc)
+    tId = cluster_template.id
+    client = admin_mc.client
+
+    cconfig = {
+        "rancherKubernetesEngineConfig": {
+            "services": {
+                "type": "rkeConfigServices",
+                "kubeApi": {
+                    "alwaysPullImages": "false",
+                    "podSecurityPolicy": "false",
+                    "serviceNodePortRange": "30000-32767",
+                    "type": "kubeAPIService"
+                }
+            }
+        },
+        "defaultPodSecurityPolicyTemplateId": "restricted",
+    }
+    azureClientId = "rancherKubernetesEngineConfig.cloudProvider.\
+azureCloudProvider.aadClientId"
+    azureClientSecret = "rancherKubernetesEngineConfig.cloudProvider.\
+azureCloudProvider.aadClientSecret"
+
+    questions = [{
+                  "variable": "dockerRootDir",
+                  "required": "true",
+                  "type": "string",
+                  "default": ""
+                 },
+                 {
+                  "variable": azureClientId,
+                  "required": "true",
+                  "type": "string",
+                  "default": "abcdClientId"
+                 },
+                 {
+                  "variable": azureClientSecret,
+                  "required": "true",
+                  "type": "string",
+                  "default": ""
+                 }]
+
+    rev = client.create_cluster_template_revision(clusterConfig=cconfig,
```

Comment:
we prolly need remove_resource for template and revision too?

Reply:
create_cluster_template will do the remove_resource on the template, which in turn deletes all the revisions.